sky/serve/replica_managers.py

```diff
@@ -69,8 +71,14 @@ def launch_cluster(task_yaml_path: str,
         config = common_utils.read_yaml(os.path.expanduser(task_yaml_path))
         if resources_override is not None:
             resource_config = config.get('resources', {})
+            # SkyServe does not support ordered resources.
+            # Remove 'any_of' as it only contains zone/region/cloud
+            if resources_override.get('use_spot', True):
+                resource_config.pop('any_of', None)
```

Comment:
We should only do this for spot policy. This could accidentally pop the vanilla spot support's any_of field. Actually why not add this override in `_get_{spot,on_demans}_resources_override_dict`?